core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -370,6 +400,62 @@ oppia.directive('questionPlayer', [
             $scope.resultsLoaded = true;
           };
 
+          var calculateMasteryDegrees = function(questionStateData) {
+            createMasteryPerSkillMapping();
+            var timesToGetLowestMastery = (
+              (MAX_MASTERY_GAIN_PER_QUESTION - MAX_MASTERY_LOSS_PER_QUESTION) /
+              WRONG_ANSWER_PENALTY_FOR_MASTERY);
+
+            for (var question in questionStateData) {
+              var questionData = questionStateData[question];
+              var totalMasteryChange = MAX_MASTERY_GAIN_PER_QUESTION;
+
+              if (!questionData.viewedSolution) {
+                if (questionData.usedHints) {
+                  totalMasteryChange -= (
+                    questionData.usedHints.length *
+                    VIEW_HINT_PENALTY_FOR_MASTERY);
+                }
+                if (questionData.answers) {
+                  if (questionData.answers.length - 1 >=
```

Comment:
Do not put this if condition

Reply:
I kept this check for setting the lowest bound, but I changed it to a different way to avoid the problem we talked about yesterday.